src/widgets/attachments/attachments-overview.component.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from "react";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import {
+  getAttachments,
+  createAttachment,
+  deleteAttachment
+} from "./attachments.resource";
+import Gallery from "react-grid-gallery";
+import styles from "./attachments-overview.css";
+
+export default function AttachmentsOverview() {
+  const [attachments, setAttachments] = useState([]);
+  const [currentImage, setCurrentImage] = useState(0);
+
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  useEffect(() => {
+    if (patientUuid) {
+      const abortController = new AbortController();
+      getAttachments(patientUuid, true, abortController).then(
+        (response: any) => {
+          const listItems = response.data.results.map(attachment => ({
+            id: `${attachment.uuid}`,
+            src: `/openmrs/ws/rest/v1/attachment/${attachment.uuid}/bytes`,
+            thumbnail: `/openmrs/ws/rest/v1/attachment/${attachment.uuid}/bytes`,
+            thumbnailWidth: 320,
+            thumbnailHeight: 212,
+            caption: attachment.comment
+          }));
+          setAttachments(listItems);
+        }
+      );
+    }
+  }, [patientUuid]);
+
+  function handleUpload(e: React.SyntheticEvent, files: FileList | null) {
+    e.preventDefault();
+    e.stopPropagation();
+    const abortController = new AbortController();
+    if (files) {
+      const attachments_tmp = attachments.slice();
+      for (let i = 0; i < files.length; i++) {
+        createAttachment(
+          patientUuid,
+          files[i],
+          files[i].name,
+          abortController
+        ).then(response => {
+          const new_attachment = {
+            id: `${response.data.uuid}`,
+            src: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
+            thumbnail: `/openmrs/ws/rest/v1/attachment/${response.data.uuid}/bytes`,
+            thumbnailWidth: 320,
+            thumbnailHeight: 212,
+            caption: response.data.comment
+          };
+          attachments_tmp.push(new_attachment);
+        });
+      }
+      setAttachments(attachments_tmp);
+    }
+  }
+
+  function handleCurrentImageChange(index) {
+    setCurrentImage(index);
+  }
+
+  function handleDelete() {
+    if (window.confirm("Are you sure you want to delete attachment?")) {
```

Comment:
I guess this is right now only for dev convenience, right?

Do we have a UX strategy how to deal with such dialogs or does that come back to the need for a modal manager @brandones ?

Reply:
This was my first instinct but I think we should have a standard for doing these kind of things.

Reply:
I fully agree. Not sure if its on our radar design-wise, but I will also consider it for the upcoming modal dialog manager RFC.

Reply:
Sure, having a technical strategy for this would be good.

I would also like us to do some thinking (with Cairan's help, if possible) about "UX strategy" in a more broad sense -- when are dialog popups really necessary, what must they contain to be worthwhile, how can they be avoided, etc.? For example, once we have a proper modal, we should display the thumbnail of the attachment being deleted in the dialog box.